src/Tools/dotnet-monitor/Egress/Extension/ExtensionEgressProvider.cs

```diff
@@ -57,5 +63,26 @@ public ExtensionEgressProvider(IEgressPropertiesProvider propertyProvider, Exten
 
             return result.ArtifactPath;
         }
+
+        private string GetConfigurationSection(string providerName, string providerType)
+        {
+            IConfigurationSection providerTypeSection = _configurationProvider.GetConfigurationSection(providerType);
+            IConfigurationSection providerNameSection = providerTypeSection.GetSection(providerName);
+
+            if (providerNameSection.Exists())
+            {
+                // Trim prefix for binding options on the extension side
+                // Format Example: {[Egress:AzureBlobStorage:monitorBlob:BlobPrefix, artifacts]} becomes {[BlobPrefix, artifacts]}
+                string prefix = FormattableString.Invariant($"{ConfigurationKeys.Egress}:{providerType}:{providerName}:");
+
+                var configAsDict = providerNameSection.AsEnumerable().ToDictionary(c => c.Key.Length > prefix.Length ? c.Key.Remove(0, prefix.Length) : c.Key.Remove(0, c.Key.Length), c => c.Value);
```

Comment:
You might need to filter out top-level properties (ones that have children) that have a null value.